app/Http/Controllers/Administration/UpdateController.php

```diff
@@ -117,23 +122,33 @@ public function view(): View
 	 * The whole code around installation/upgrade/migration should
 	 * thoroughly be revised an refactored.
 	 */
-	public function migrate(Request $request): View
+	public function migrate(Request $request): View|Response
 	{
-		if (
-			AccessControl::is_admin() || AccessControl::noLogin() ||
-			AccessControl::log_as_admin($request['username'] ?? '', $request['password'] ?? '', $request->ip())
-		) {
+		// This conditional code makes use of lazy boolean evaluation: a || b does not execute b if a is true.
+		// 1. We check if we are logged in
+		// 2. if not, we check if the admin user is registered (if not login in such case)
+		// 3. We have an admin user set up, try to login with Legacy method: hash(username) + hash(password).
+		// 4. if Legacy login failed, try to login the normal way.
```

Comment:
I don't like "we" in comments. Who is "we"? The developers, probably not.

```suggestion
		// This conditional code makes use of lazy boolean evaluation: a || b does not execute b if a is true.
		// 1. Check whether the user is already logged in properly
		// 2. Check if the admin user is registered and login as admin, if not
		// 3. Attempt to login as an admin user using the legacy method: hash(username) + hash(password).
		// 4. Try to login the normal way.
		//
		// TODO: Step 2 will become unnecessary once admin registration has become part of the installation routine; after that the case that no admin is registered cannot occur anymore
		// TODO: Step 3 will become unnecessary once the admin user of any existing installation has at least login once and the admin user has therewith migrated to use a non-hashed user name
```

Also more information _WHY_ we need these steps.